deps/reftable/system.h

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2020 Google LLC
+
+Use of this source code is governed by a BSD-style
+license that can be found in the LICENSE file or at
+https://developers.google.com/open-source/licenses/bsd
+*/
+
+#ifndef SYSTEM_H
+#define SYSTEM_H
+
+#if REFTABLE_IN_GITCORE
+
+#include "git-compat-util.h"
+#include <zlib.h>
+
+#else
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <zlib.h>
```

Comment:
This currently doesn't compile on Windows due to <sys/time.h> not being found. I think it should work just fine if changing this to `<time.h>`. Instead, we could probably have the equivalent of `REFTABLE_IN_GITCORE`, but I think I'd rather prefer the header to be agnostic of libgit2.

Reply:
On linux, this would yield: 

c/stack.c:226:12: error: implicit declaration of function 'gettimeofday' [-Werror=implicit-function-declaration]
  226 |  int err = gettimeofday(&deadline, NULL);
      |            ^~~~~~~~~~~~